src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -955,8 +955,12 @@ let AdjustValSynInfoInSignature g ty (SynValInfo(argsData, retData) as sigMD) =
     | _ ->
         sigMD
 
+let getArgInfoCache =
+    let options = Caches.CacheOptions.getDefault() |> Caches.CacheOptions.withNoEviction
+    let factory _ = new Caches.Cache<_, ArgReprInfo>(options, "argInfoCache")
+    WeakMap.getOrCreate factory
```

Comment:
Are these caches going to be cleared when asking FCS to clear its caches?

Reply:
Currently not, I think since they're weakly attached to higher lever stuff, they'll just get GC'd with it. (?)